receiver/postgresqlreceiver/scraper.go

```diff
@@ -0,0 +1,284 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresqlreceiver
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/postgresqlreceiver/internal/metadata"
+)
+
+type postgreSQLScraper struct {
+	logger *zap.Logger
+	config *Config
+}
+
+func newPostgreSQLScraper(
+	logger *zap.Logger,
+	config *Config,
+) *postgreSQLScraper {
+	return &postgreSQLScraper{
+		logger: logger,
+		config: config,
+	}
+}
+
+var initializeClient = func(p *postgreSQLScraper, database string) (client, error) {
```

Comment:
This should not be a variable but a defined function

Reply:
This was required for mocking. I'll see if I can work around this when reimplementing with testify/mock.